Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs

```diff
@@ -254,42 +358,63 @@ private static void SearchForAndDisableExtraPlayspaces(IEnumerable<GameObject> r
             // Our task is to search for any additional play spaces that may have been loaded, and disable them.
             foreach (GameObject rootGameObject in rootGameObjects)
             {
-                if (rootGameObject == mixedRealityPlayspace.gameObject)
-                {   // Don't disable our existing playspace
-                    continue;
+                if (rootGameObject.name.Equals(NameEnabled) || rootGameObject.name.Equals(NameDisabled))
+                {
+                    MixedRealityPlayspace playspace = rootGameObject.EnsureComponent<MixedRealityPlayspace>();
                 }
 
-                if (rootGameObject.name.Equals(Name))
+                foreach (MixedRealityPlayspace playspace in rootGameObject.GetComponentsInChildren<MixedRealityPlayspace>(true))
```

Comment:
Instead of searching for this script, can't we continue making the current assumption about camera parents? Or naming? Both of which seem to still be enforced by this script.

Reply:
That's the point of this PR. Searching by name or parent is no longer reliable when you're dealing with multiple scenes and multiple main cameras.

Reply:
But you're still using the main camera to [ensure](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4455/files#diff-a2a2bab0eb44c8a46d3c07c490881396R231) or [set](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4455/files#diff-a2a2bab0eb44c8a46d3c07c490881396R214) the playspace.

Reply:
You're right, it shouldn't do that unless the camera system is disabled. I'm pushing a change with cleaner logic.